bumble/transport/pyusb.py

```diff
@@ -308,3 +328,73 @@ def device_path(device):
     packet_sink.start()
 
     return UsbTransport(device, packet_source, packet_sink)
+
+
+def _power_cycle(device: UsbDevice) -> UsbDevice:
+    """
+    For devices connected to compatible USB hubs: Performs a power cycle on a given USB device.
+    This involves temporarily disabling its port on the hub and then re-enabling it.
+    """
+    device_path = f'{device.bus}-{".".join(map(str, device.port_numbers))}'  # type: ignore
+    hub = _find_hub_by_device_path(device_path)
+
+    if hub:
+        try:
+            device_port = device.port_numbers[-1]  # type: ignore
+            _set_port_status(hub, device_port, False)
+            time.sleep(POWER_CYCLE_DELAY)
```

Comment:
Blocking sleeps aren't good for async code. This function should become an async function, and use `asyncio.sleep()` instead.

Reply:
Agree, done.